gradle-publish-plugin not working anymore so can't make a release

```
> Task :closeRepository
DEPRECATION WARNING. The staging repository ID is not provided. The fallback mode may impact release reliability and is deprecated. Please consult the project FAQ how it can be fixed.
Attempt 21/21 failed. RepositoryInTransitionException was thrown with message 'Repository 'iominio-1094' (in 'open' state) is in transition. Check again later.'. Giving up. Configure longer timeout if necessary.

> Task :closeRepository FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':closeRepository'.
> Repository 'iominio-1094' (in 'open' state) is in transition. Check again later.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.2.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 47s
1 actionable task: 1 executed
```

Related issues

- https://github.com/Codearte/gradle-nexus-staging-plugin/issues/127
- https://github.com/Codearte/gradle-nexus-staging-plugin/issues/77

https://solidsoft.wordpress.com/2019/02/22/reliable-releasing-to-maven-central-from-travis-using-gradle-2019-edition/

We need to move to 

```
plugins {
    ... //other plugins used in your project
    id 'io.codearte.nexus-staging' version '0.20.0'
    id 'de.marcphilipp.nexus-publish' version '0.2.0'
}
 
publishing {
    ... //your current publishing to Maven Central configuration
}
 
//optionally
nexusStaging {
    packageGroup = "your-package-group-if-different-than-groupId"
}
 
//optionally
nexusPublishing {
    //for custom configuration if needed - credentials
    //       are by default taken from nexus-staging
    //       or from properties nexusUsername and nexusPassword
}
```

--- a/build.gradle
+++ b/build.gradle
@@ -21,7 +21,8 @@
 plugins {
   id "com.github.johnrengelman.shadow" version "5.2.0"
   id "com.github.spotbugs" version "4.0.1"
-  id "io.codearte.nexus-staging" version "0.21.2"
+  id "io.codearte.nexus-staging" version "0.22.0"
+  id "de.marcphilipp.nexus-publish" version "0.4.0"
   id "com.diffplug.gradle.spotless" version "4.4.0"
 }
 
@@ -29,10 +30,11 @@ plugins {
  * Root project definitions
  */
 apply plugin: 'io.codearte.nexus-staging'
+apply plugin: "de.marcphilipp.nexus-publish"
 
 allprojects {
     group = 'io.minio'
-    version = '7.1.1'
+    version = '7.1.3'
     if (!project.hasProperty('release')) {
         version += '-DEV'
     }
@@ -44,6 +46,7 @@ subprojects {
     apply plugin: "com.diffplug.gradle.spotless"
 
     repositories {
+        mavenLocal()
         mavenCentral()
     }
 
@@ -93,7 +96,7 @@ subprojects {
 
     check.dependsOn localeTest
 
-    sourceCompatibility = 1.8
+    sourceCompatibility = JavaVersion.VERSION_1_8
 
     spotless {
         java {
@@ -109,16 +112,14 @@ subprojects {
 }
 
 project(':api') {
-    apply plugin: 'com.github.johnrengelman.shadow'
+    apply plugin: 'java'
     apply plugin: 'maven'
+    apply plugin: 'maven-publish'
     apply plugin: 'signing'
+    apply plugin: 'com.github.johnrengelman.shadow'
+    apply plugin: "de.marcphilipp.nexus-publish"
 
     archivesBaseName = 'minio'
-
-    nexusStaging {
-        packageGroup = group
-        stagingProfileId = '9b746c9f8abc1'
-    }
 
     configurations.all {
         resolutionStrategy {
@@ -166,58 +167,64 @@ project(':api') {
         archives javadocJar, sourcesJar, shadowJar
     }
 
-    signing {
-        if (project.properties.containsKey('signing.keyId')) {
-            sign configurations.archives
-        }
-    }
-
-    uploadArchives {
-        repositories {
-            mavenDeployer {
-                beforeDeployment {
-                    MavenDeployment deployment -> signing.signPom(deployment)
-                }
-
-                ext.nexusUsername = project.properties['nexusUsername']
-                ext.nexusPassword = project.properties['nexusPassword']
-
-                repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
-                    authentication(userName: ext.nexusUsername, password: ext.nexusPassword)
-                }
-
-                snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
-                    authentication(userName: ext.nexusUsername, password: ext.nexusPassword)
-                }
-
-                pom.project {
-                    name 'minio'
-                    packaging 'jar'
-                    description 'MinIO Java SDK for Amazon S3 Compatible Cloud Storage'
-                    url 'https://github.com/minio/minio-java'
-                    inceptionYear '2015'
+    publishing {
+        publications {
+            mavenJava(MavenPublication) {
+                artifactId archivesBaseName
+                from components.java
+                artifact sourcesJar
+                artifact shadowJar
+                artifact javadocJar
+                pom {
+                    name = 'minio'
+                    packaging = 'jar'
+                    description = 'MinIO Java SDK for Amazon S3 Compatible Cloud Storage'
+                    url = 'https://github.com/minio/minio-java'
+                    inceptionYear = '2015'
 
                     scm {
-                        connection 'scm:git:git@github.com:minio/minio-java.git'
-                        developerConnection 'scm:git:git@github.com:minio/minio-java.git'
-                        url 'http://github.com/minio/minio-java'
+                        connection = 'scm:git:git@github.com:minio/minio-java.git'
+                        developerConnection = 'scm:git:git@github.com:minio/minio-java.git'
+                        url = 'http://github.com/minio/minio-java'
                     }
 
                     licenses {
                         license {
-                            name 'The Apache License, Version 2.0'
-                            url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                            name = 'The Apache License, Version 2.0'
+                            url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
                         }
                     }
 
                     developers {
                         developer {
-                            id 'minio'
-                            name 'MinIO Inc.'
-                            email 'dev@min.io'
+                            id = 'minio'
+                            name = 'MinIO Inc.'
+                            email = 'dev@min.io'
                         }
                     }
                 }
+            }
+        }
+    }
+
+    signing {
+        if (project.properties.containsKey('signing.keyId')) {
+            sign publishing.publications.mavenJava
+        }
+    }
+
+    nexusStaging {
+        packageGroup = group
+        stagingProfileId = '9b746c9f8abc1'
+        username = project.properties['nexusUsername']
+        password = project.properties['nexusPassword']
+    }
+
+    nexusPublishing {
+        repositories {
+            sonatype {
+                username = project.properties['nexusUsername']
+                password = project.properties['nexusPassword']
             }
         }
     }
@@ -231,10 +238,10 @@ project(':examples') {
 
     sourceSets {
         main {
-	    java {
-	        srcDirs = ["$rootDir/examples"]
-	    }
-	}
+            java {
+                srcDirs = ["$rootDir/examples"]
+            }
+        }
     }
 }
 
@@ -281,6 +288,6 @@ project(':functional') {
         jvmArgs "-Xms256m"
         if (rootProject.hasProperty('endpoint') || OperatingSystem.current().isWindows()) {
             args = [ext.endpoint, ext.accessKey, ext.secretKey, ext.region]
-	}
+        }
     }
 }

